### PR TITLE
Post API modification and bug fixes

### PR DESCRIPTION
**Pull request by FalseChord**
### Post API modifications
1. Insert Post API
- Now accept "cards" as a parameter, which contains an array of card objects.
- Any cards that have id will be ignored.
2. Update Post API
- Now accept "cards" as a parameter, which contains an array of the card object
- Cards that have id will be updated.
- Cards that have no id will be inserted.
- Cards belong to the post but not exist in the cards parameter will be deleted ( set active to 0 ).
- If "cards" parameter not presented, all cards belong to the post will be deleted.
3. Refactor
- Refactor the handlers for post publishing and updating
4. Example
- Please refer to test file: routes/post_test.go
- For insert post with cards: search for "WithMultipleCards"
- For update post with cards: search for "UpdateCards"

### Bug fixes
1. Inserting or updating a post now update the update time of the project the post belongs to (if there is one).

### CODE REVIEW

File: pkg/cards/sql.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+			cardsToBeUpdated = append(cardsToBeUpdated, card)
+		}
+	}
+
+	if postID != 0 {
+		var cardIDs []uint32
+		for _, card := range cardsToBeUpdated {
+			cardIDs = append(cardIDs, card.ID)
+		}
+		stmt, err := buildDeleteStmt(postID, cardIDs)
```
Comment (ichiaohsu) [inline pkg/cards/sql.go]: I'm not sure why we have to delete cards id if post id is not 0

Comment (FalseChord) [reply pkg/cards/sql.go]: See line 75, the query specifies to delete cards not in the list of cards to be updated.
This means that cards not contained in the parameter `cards` and exists in the DB in the same time should be deleted, according to the spec.

**Comment by FalseChord** [comment]
> I found I did not get why TrasactionIDPlaceholder works. Is it SQL-injection free?

It's protected from SQL-injection, since we replace the placeholder (defined in config file) by the ID returned by calling the inserted ID (returned by LastInsertId()), there is no place for user-generated content in this process.

**Comment by ichiaohsu** [comment]
> > I found I did not get why TrasactionIDPlaceholder works. Is it SQL-injection free?
> 
> It's protected from SQL-injection, since we replace the placeholder (defined in config file) by the ID (returned by LastInsertId()), there is no place for user-generated content in this process.

Then I don't have any further question. Feel free to merge this.

**Pull request merged**